### PR TITLE
build: consolidate divergent forks

### PR DESCRIPTION
**Pull request by stvstnfrd**
Somewhere along the line, this repo's `HEAD` got pointed _away_ from `master` and _to_ `feanil/python-modernize` .

Since then, most (mostly human) PRs have merged to `HEAD`, `feanil/python-modernize`, while some PRs (mostly bots) have merged to `master`.

This can be explained by each flow's default behavior:
- the Github web UI defaults PRs to `HEAD`
- scripts are often (hard-)coded to assume `HEAD` is `master`.

This PR combines the two timelines, as well as again updating dependencies. After this merges, I intend to:
- fast-forward `feanil/python-modernize` to match the new `master`
- reset `HEAD` to `master`
- update our deploy scripts/requirements to ensure we are deploying from `master` (or derivative).

### CODE REVIEW

**Comment by sarina** [comment]
See https://openedx.atlassian.net/browse/CENG-104 - we should decide what the right course of action is here

**Comment by stvstnfrd** [comment]
@sarina 
TL;DR: I think this should land regardless of CENG-104.

From my understanding of that ticket, each discusses a different aspect:
- CENG-104 is about which _repository_ to install from; our fork or upstream?
- This PR is about which _branch_ on our own fork to install from; `master` or feanil's personal feature branch?

Within our own repository, developers and automation are merging code to different branches, depending on their workflow [2], so we now have a fork _within_ a fork [1]. This PR consolidates the schism within _edx_'s tree, not that between `edx` and `schoolyourself`.

This should probably even be considered a prerequisite for CENG-104 anyway; before we compare our fork against theirs, we should ensure our changes aren't spread across multiple branches.

There should be no user-facing effects of these proposed changes; all changes (to both branches) appear to have been build/test related.

- [1] and since `CODEOWNERS` only existed on the feature branch and _not_ `master`, @edx/community-engineering wasn't being tagged on any PR against `master`, like security upgrades, etc. Also, the security PRs weren't going into the default/`HEAD` branch.
- [2] this happened to me locally where one of my `git` scripts choked on the non-default branch name

**Comment by sarina** [comment]
Got it, that makes sense. I read too fast.

**Review by sarina — APPROVED** [review]
